Description: Upstream changes introduced in version 8.09.10beta1-1
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 openpsa2 (8.09.10beta1-1) unstable; urgency=low
 .
   * Initial release
 .
 The person named in the Author field signed this changelog entry.
Author: Henri Bergius <henri.bergius@iki.fi>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- /dev/null
+++ openpsa2-8.09.10beta1/lintian-complains.txt
@@ -0,0 +1,210 @@
+E: openpsa2: changelog-file-not-compressed changelog.Debian
+N: 
+N:    Changelog files should be compressed using "gzip -9". Even if they start
+N:    out small, they will become large with time.
+N:    
+N:    Refer to Debian Policy Manual section 12.7 (Changelog files) for
+N:    details.
+N:    
+N:    Severity: important, Certainty: certain
+N: 
+W: openpsa2: new-package-should-close-itp-bug
+N: 
+N:    This package appears to be the first packaging of a new upstream
+N:    software package (there is only one changelog entry and the Debian
+N:    revision is 1), but it does not close any bugs. The initial upload of a
+N:    new package should close the corresponding ITP bug for that package.
+N:    
+N:    This warning can be ignored if the package is not intended for Debian or
+N:    if it is a split of an existing Debian package.
+N:    
+N:    Refer to Debian Developer's Reference section 5.1 (New packages) for
+N:    details.
+N:    
+N:    Severity: normal, Certainty: certain
+N: 
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midgard/admin/asgard/documentation/asgard_midgard_topic.fi.txt
+N: 
+N:    This executable file is not an ELF format binary, and does not start
+N:    with the #! sequence that marks interpreted scripts. It might be a sh
+N:    script that fails to name /bin/sh as its shell, or it may be incorrectly
+N:    marked as executable. Sometimes upstream files developed on Windows are
+N:    marked unnecessarily as executable on other systems.
+N:    
+N:    If you are using debhelper to build your package, running dh_fixperms
+N:    will often correct this problem for you.
+N:    
+N:    Refer to Debian Policy Manual section 10.4 (Scripts) for details.
+N:    
+N:    Severity: normal, Certainty: certain
+N: 
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/net/nemein/tag/tag.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/exec/fckeditor/fb_connectors_util.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midgard/admin/asgard/config/schemadb_permissions.inc
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midgard/admin/asgard/config/schemadb_parameter.inc
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/exec/fckeditor/fb_connectors_commands.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/widget/video.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midgard/admin/asgard/config/schemadb_config.inc
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/services/i18n.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/net/nehmer/blog/config/schemadb_link.inc
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/interviews/viewer.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/widget/tabledata.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/mypage/style/show-workingon-xml.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midgard/admin/asgard/exec/test.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/misc.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendar/style/show-event-raw.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendarwidget/style/calendarwidget-day-event.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/admin/babel/plugin.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendarwidget/style/calendarwidget-month-footer.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midgard/admin/asgard/midcom/interfaces.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midgard/admin/asgard/documentation/asgard_midgard_topic.en.txt
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/mypage/documentation/CHANGES
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/net/nehmer/blog/link.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/net/nehmer/markdown/locale/default.pl.txt
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/services/_sessioning.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/projects/config/midcom_dba_classes.inc
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendar/style/show-day-footer.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/net/nehmer/static/link.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midgard/admin/asgard/config/schemadb_parameters.inc
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendarwidget/style/calendarwidget-day-events-footer.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/services/at/cron/clean.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/services/session.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/directmarketing/style/show-message-copy-ok.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/net/nehmer/static/style/admin-delete-link.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/routamc/positioning/importer.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/net/nehmer/blog/config/midcom_dba_classes.inc
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/_styleloader.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/widget/colorpicker.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/net/nehmer/static/config/midcom_dba_classes.inc
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/exec/fckeditor/fb_connectors_connector.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/admin/settings/documentation/CHANGES
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/admin/babel/style/midcom_admin_babel_edit.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/admin/babel/style/midcom_admin_babel_select.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/exec/fckeditor/fb_connectors_config.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/directmarketing/config/schemadb_default_message_copy.inc
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/net/nemein/wiki/documentation/markdown.en.txt
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/net/nehmer/blog/style/admin-delete-link.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendarwidget/style/calendarwidget-week-start.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/openpsa2/schema/net_nehmer_static.xml
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midgard/admin/asgard/documentation/asgard_midgard_topic.pl.txt
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/services/_i18n_l10n.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/net/nehmer/blog/style/admin-create-link.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendar/style/show-day-header.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendarwidget/style/calendarwidget-day-events-header.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/routamc/positioning/locale/default.sv.txt
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/directmarketing/style/show-message-copy.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/jabber/style/show-userinfo.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/configuration.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/openpsa2/schema/net_nehmer_blog.xml
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/mypage/midcom/interfaces.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/routamc/positioning/locale/default.fi.txt
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/routamc/positioning/exec/manual-position.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/routamc/positioning/locale/default.en.txt
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/mypage/style/show-userinfo.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/contacts/style/show-buddylist-header.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/type/tabledata.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendarwidget/style/calendarwidget-day-header.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendarwidget/style/calendarwidget-week-end.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/exec/fckeditor/fb_connectors_basexml.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendar/style/show-day-item.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/xsspreventer/static/xsspreventer.png
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/sales/style/show-deliverable-edit.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/notifications/documentation/CHANGES
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midgard/admin/asgard/config/schemadb_default.inc
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midgard/admin/asgard/config/schemadb_preferences.inc
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendarwidget/style/calendarwidget-day-footer.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/mailtemplate.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/admin/folder/style/midcom-admin-show-folder-actions.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/constants.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/routamc/positioning/log.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/contacts/buddy.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/calendarwidget/style/calendarwidget-month-header.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/routamc/positioning/locale/default.pl.txt
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/org/openpsa/contacts/style/show-buddylist-footer.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/debug.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/nav.php
+W: openpsa2: executable-not-elf-or-script ./usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/exec/fckeditor/fb_connectors_io.php
+W: openpsa2: maintainer-script-ignores-errors postinst
+N: 
+N:    The maintainer script doesn't seem to set the -e flag which ensures that
+N:    the script's execution is aborted when any executed command fails.
+N:    
+N:    Refer to Debian Policy Manual section 10.4 (Scripts) for details.
+N:    
+N:    Severity: normal, Certainty: certain
+N: 
+W: openpsa2: extra-license-file usr/share/openpsa2/www/jQuery/jquery.ui-1.8.2/GPL-LICENSE.txt
+N: 
+N:    All license information should be collected in the debian/copyright
+N:    file. This usually makes it unnecessary for the package to install this
+N:    information in other places as well.
+N:    
+N:    Refer to Debian Policy Manual section 12.5 (Copyright information) for
+N:    details.
+N:    
+N:    Severity: normal, Certainty: possible
+N: 
+W: openpsa2: extra-license-file usr/share/openpsa2/www/jQuery/jquery.ui-1.8.2/MIT-LICENSE.txt
+W: openpsa2: extra-license-file usr/share/openpsa2/www/jQuery/ui/GPL-LICENSE.txt
+W: openpsa2: extra-license-file usr/share/openpsa2/www/jQuery/ui/MIT-LICENSE.txt
+W: openpsa2: embedded-javascript-library usr/share/openpsa2/www/midcom.helper.datamanager2/colorpicker/jquery.js
+N: 
+N:    This package contains an embedded copy of JavaScript libraries that are
+N:    now available in their own packages (for example, JQuery, Prototype,
+N:    Mochikit or "Cropper"). Please depend on the appropriate package and
+N:    symlink the library into the appropriate location.
+N:    
+N:    Refer to Debian Policy Manual section 4.13 (Convenience copies of code)
+N:    for details.
+N:    
+N:    Severity: normal, Certainty: possible
+N: 
+W: openpsa2: embedded-javascript-library usr/share/openpsa2/www/midcom.helper.datamanager2/fckeditor/fckeditor.js
+W: openpsa2: extra-license-file usr/share/openpsa2/www/midcom.helper.datamanager2/fckeditor/license.txt
+W: openpsa2: extra-license-file usr/share/openpsa2/www/midcom.helper.datamanager2/tinymce/license.txt
+W: openpsa2: embedded-javascript-library usr/share/openpsa2/www/midcom.helper.datamanager2/tinymce/tiny_mce.js
+W: openpsa2: embedded-javascript-library usr/share/openpsa2/www/midcom.helper.datamanager2/tinymce/tiny_mce_popup.js
+W: openpsa2: embedded-javascript-library usr/share/openpsa2/www/midcom.helper.datamanager2/tinymce/tiny_mce_src.js
+I: openpsa2: font-in-non-font-package usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/widget/captcha-font.ttf
+N: 
+N:    This package contains a *.ttf, *.otf, or *.pfb file, file extensions
+N:    used by TrueType, OpenType, or Type 1 fonts, but the package does not
+N:    appear to be a dedicated font package. Dedicated font package names
+N:    should begin with ttf-, otf-, or t1-, depending on the types of fonts
+N:    included. (Type 1 fonts are also allowed in packages starting with
+N:    xfonts-.) If the font is already packaged, you should depend on that
+N:    package instead. Otherwise, normally the font should be packaged
+N:    separately, since fonts are usually useful outside of the package that
+N:    embeds them.
+N:    
+N:    Severity: wishlist, Certainty: possible
+N: 
+W: openpsa2: embedded-php-library usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/widget/markdown.php
+N: 
+N:    This package appears to contain an embedded copy of a PHP library.
+N:    Please depend on the respective package providing the library and make
+N:    sure it can be found by the scripts via the include_path.
+N:    
+N:    Refer to Debian Policy Manual section 4.13 (Convenience copies of code)
+N:    for details.
+N:    
+N:    Severity: normal, Certainty: possible
+N: 
+W: openpsa2: extra-license-file usr/share/php5/openpsa2/lib/midcom/helper/datamanager2/widget/recaptcha/LICENSE
+W: openpsa2: extra-license-file usr/share/php5/openpsa2/lib/net/nehmer/markdown/documentation/licence.en.txt
+W: openpsa2: embedded-php-library usr/share/php5/openpsa2/lib/net/nehmer/markdown/lib/markdown.php
+W: openpsa2: embedded-php-library usr/share/php5/openpsa2/lib/net/nehmer/markdown/markdown.php
+W: openpsa2: embedded-php-library usr/share/php5/openpsa2/lib/net/nemein/rss/magpierss/rss_parse.inc
+I: openpsa2: package-contains-empty-directory usr/share/openpsa2/views/
+N: 
+N:    This package installs an empty directory. This might be intentional but
+N:    it's normally a mistake. If it is intentional, add a lintian override.
+N:    
+N:    If a package ships with or installs empty directories, you can remove
+N:    them in debian/rules by calling:
+N:    
+N:     $ find path/to/base/dir -type d -empty -delete
+N:    
+N:    Severity: wishlist, Certainty: possible
+N: 
--- openpsa2-8.09.10beta1.orig/quick_init.php
+++ openpsa2-8.09.10beta1/quick_init.php
@@ -1,27 +1,29 @@
 <?php
-if (count($argv) != 2)
-{
-    die("Usage: php quick_init.php midgardconffile\n");
-}
-
 if (!extension_loaded('midgard2'))
 {
     die("Midgard2 is not installed in your PHP environment.\n");
 }
 
-// Create a config file
 $config = new midgard_config();
-$config->dbtype = 'SQLite';
-$config->database = $argv[1];
-$config->databasedir = '/tmp';
-$config->tablecreate = true;
-$config->tableupdate = true;
-$config->loglevel = 'debug';
-if (!$config->save_file($argv[1], false))
+if (!$config->read_file('openpsa2', false))
 {
-    die("Failed to save Midgard2 config file");
+    // Create a config file
+    $config = new midgard_config();
+    $config->dbtype = 'SQLite';
+    $config->database = $argv[1];
+    $config->dbdir = '/var/lib/openpsa2';
+    $config->blobdir = '/var/lib/openpsa2/blobs';
+    $config->sharedir = '/usr/share/openpsa2';
+    $config->vardir = '/var/lib/openpsa2';
+    $config->cachedir = '/var/cache/openpsa2';
+    $config->logfilename = '/var/log/openpsa2/midgard.log';
+    $config->loglevel = 'debug';
+    if (!$config->save_file('openpsa2', false))
+    {
+        die("Failed to save Midgard2 config file to /etc/midgard2/conf.d\n");
+    }
+    echo "Configuration file /etc/midgard2/conf.d/openpsa2 created.\n";
 }
-echo "Configuration file /etc/midgard2/conf.d/{$argv[1]} created.\n";
 
 // Open a DB connection with the config
 $midgard = midgard_connection::get_instance();
@@ -36,8 +38,17 @@ if (!$config->create_blobdir())
 }
 
 // Create storage
-midgard_storage::create_base_storage();
-echo "Database initialized, preparing storage for MgdSchema classes:\n";
+if (!midgard_storage::create_base_storage())
+{
+    if ($midgard->get_error_string() != 'MGD_ERR_OK')
+    {
+        die("Failed to create base database structures" . $midgard->get_error_string() . "\n");
+    }
+}
+else
+{
+    echo "Database initialized, preparing storage for MgdSchema classes:\n";
+}
 
 $re = new ReflectionExtension('midgard2');
 $classes = $re->getClasses();
@@ -55,6 +66,7 @@ foreach ($classes as $refclass)
     $type = $refclass->getName();
             
     midgard_storage::create_class_storage($type);
+    midgard_storage::update_class_storage($type);
     echo "  Created storage for {$type}\n";
 }
 ?>
--- openpsa2-8.09.10beta1.orig/rootfile.php
+++ openpsa2-8.09.10beta1/rootfile.php
@@ -17,6 +17,8 @@ ini_set('memory_limit', '68M');
 
 // Path to the MidCOM environment
 define('MIDCOM_ROOT', realpath(dirname(__FILE__)) . '/lib');
+define('MIDCOM_STATIC_ROOT', '/openpsa2-static');
+define('OPENPSA2_PREFIX', dirname($_SERVER['SCRIPT_NAME']));
 
 // Include Midgard1 compatibility APIs needed for running OpenPSA under Midgard2
 require(MIDCOM_ROOT . '/ragnaroek-compat.php');
@@ -28,7 +30,7 @@ openpsa_prepare_superglobal();
 
 $GLOBALS['midcom_config_local'] = array();
 $GLOBALS['midcom_config_local']['log_level'] = 5;
-$GLOBALS['midcom_config_local']['log_filename'] = '/tmp/openpsa-midcom.log';
+$GLOBALS['midcom_config_local']['log_filename'] = dirname(midgard_connection::get_instance()->config->logfilename) . '/midcom.log';
 $GLOBALS['midcom_config_local']['midcom_root_topic_guid'] = openpsa_prepare_topics();
 $GLOBALS['midcom_config_local']['midcom_services_rcs_enable'] = false;
 $GLOBALS['midcom_config_local']['person_class'] = 'openpsa_person';
@@ -45,7 +47,9 @@ $template = mgd_preparse('<(ROOT)>');
 $template_parts = explode('<(content)>', $template);
 eval('?>' . $template_parts[0]);
 $_MIDCOM->content();
-eval('?>' . $template_parts[1]);
-
+if (isset($template_parts[1]))
+{
+    eval('?>' . $template_parts[1]);
+}
 $_MIDCOM->finish();
 ?>
--- openpsa2-8.09.10beta1.orig/lib/ragnaroek-compat.php
+++ openpsa2-8.09.10beta1/lib/ragnaroek-compat.php
@@ -148,11 +148,11 @@ function openpsa_prepare_superglobal()
     // URLs and request path
     $url_components = parse_url("http://{$_SERVER['HTTP_HOST']}{$_SERVER['REQUEST_URI']}");
     $_MIDGARD['uri'] = $url_components['path'];
-    $_MIDGARD['self'] = '/';
+    $_MIDGARD['self'] = OPENPSA2_PREFIX . '/';
     $_MIDGARD['prefix'] = substr($_MIDGARD['self'], 0, -1);
 
     $_MIDGARD['argv'] = array();
-    $path_parts = explode('/', $_MIDGARD['uri']);
+    $path_parts = explode('/', substr($_MIDGARD['uri'], strlen($_MIDGARD['prefix'])));
     foreach ($path_parts as $part)
     {
         if (empty($part))
--- /dev/null
+++ openpsa2-8.09.10beta1/config/php.ini
@@ -0,0 +1,5 @@
+[midgard2]
+midgard.engine = On
+midgard.http = On
+midgard.configuration_file="/etc/midgard2/conf.d/openpsa2"
+midgard.superglobals_compat = On
--- /dev/null
+++ openpsa2-8.09.10beta1/config/lighttpd.conf
@@ -0,0 +1,26 @@
+## OpenPSA2 configuration for Lighttpd
+
+server.modules += ( 
+    "mod_fastcgi",
+    "mod_rewrite",
+    "mod_alias"
+)
+
+fastcgi.server = ( 
+    ".php" => (
+        (
+            "bin-path" => "/usr/bin/php-cgi",
+            "socket" => "/tmp/php.socket"
+        )
+    )
+)
+
+alias.url = ( 
+    "/openpsa2/" => "/usr/share/php5/openpsa2/",
+    "/openpsa2-static/" => "/usr/share/openpsa2/www/"
+)
+
+url.rewrite-once = ( 
+    "^/openpsa2-static/(.*)$" => "/openpsa2-static/$1",
+    "^(.*)\.*" => "/openpsa2/rootfile.php"
+)
--- /dev/null
+++ openpsa2-8.09.10beta1/config/MidgardObjects.xml
@@ -0,0 +1,138 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Schema xmlns="http://www.midgard-project.org/repligard/1.4">
+    <type name="midgard_person" table="person">
+        <property name="id" type="unsigned integer" primaryfield="id">
+            <description>Local non-replication-safe database identifier</description>
+        </property>
+        <property name="firstname" type="string" index="yes" unique="yes">
+            <description>First name of the person</description>
+        </property>
+        <property name="lastname" type="string" index="yes">
+            <description>Last name of the person</description>
+        </property>
+    </type>
+
+    <type name="midgard_attachment" table="blobs">
+        <property name="id" type="unsigned integer" primaryfield="id">
+            <description>Local non-replication-safe database identifier</description>
+        </property>
+        <property name="name"     type="string" index="yes">
+            <description>Filename of the attachment</description>
+        </property>
+        <property name="title"    type="string">
+            <description>Title of the attachment</description>
+        </property>
+        <property name="location" type="string" index="yes">
+            <description>Location of the attachment in the blob directory structure</description>
+        </property>
+        <property name="mimetype" type="string" index="yes">
+            <description>MIME type of the attachment</description>
+        </property>
+        <property name="parentguid" type="guid" field="parent_guid" parentfield="parent_guid">
+            <description>GUID of the object the attachment is attached to</description>
+        </property>
+    </type>
+
+    <type name="midgard_parameter" table="record_extension">
+        <property name="id" type="unsigned integer" primaryfield="id">
+            <description>Local non-replication-safe database identifier</description>
+        </property>
+        <property name="domain"   type="string" index="yes">
+            <description>Namespace of the parameter</description>
+        </property>
+        <property name="name"     type="string" index="yes">
+            <description>Key of the parameter</description>
+        </property>
+        <property name="value"    type="text">
+            <description>Value of the parameter</description>
+        </property>
+        <property name="parentguid" type="guid" field="parent_guid" parentfield="parent_guid">
+            <description>GUID of the object the parameter extends</description>
+        </property>
+    </type>
+
+    <type name="midgard_snippetdir" table="snippetdir">
+        <property name="id" type="unsigned integer" primaryfield="id">
+            <description>Local non-replication-safe database identifier</description>
+        </property>
+        <property name="name" type="string" index="yes" unique="yes">
+            <description>Path name of the snippetdir</description>
+        </property>
+        <property name="up" type="unsigned integer" link="midgard_snippetdir:id" upfield="up">
+            <description>Snippetdir the snippetdir is under</description>
+        </property>
+    </type>
+
+    <type name="midgard_snippet" table="snippet" parent="midgard_snippetdir">
+        <property name="id" type="unsigned integer" primaryfield="id">
+            <description>Local non-replication-safe database identifier</description>
+        </property>
+        <property name="name" type="string" index="yes" unique="yes">
+            <description>Path name of the snippet</description>
+        </property>
+        <property name="snippetdir" type="unsigned integer" link="midgard_snippetdir:id" parentfield="snippetdir">
+            <description>Snippetdir the snippet is under</description>
+        </property>
+        <property name="code" type="text">
+            <description>Code of the snippet</description>
+        </property>
+        <property name="doc" type="text">
+            <description>Documentation of the snippet</description>
+        </property>
+    </type>
+
+    <type name="midgard_quota" table="quota">
+        <property name="id" type="unsigned integer" primaryfield="id">
+            <description>Local non-replication-safe database identifier</description>
+        </property>
+        <property name="sgsizelimit" type="integer" field="limit_sg_size">
+            <description>Quota for the sitegroup (in bytes)</description>
+        </property>
+        <property name="sgsize"       type="integer"     field="sg_size">
+            <description>Disk usage of the sitegroup (in bytes)</description>
+        </property>
+        <property name="sgrecordslimit" type="integer" field="limit_sg_records">
+            <description>Limit of number of records for the sitegroup</description>
+        </property>
+        <property name="sgrecords"     type="integer"     field="sg_records">
+            <description>Number of records for the sitegroup</description>
+        </property>
+        <property name="typename"     type="string" index="yes">
+            <description>MgdSchema type the quota applies to</description>
+        </property>
+        <property name="typesizelimit" type="integer" field="limit_type_size">
+            <description>Quota of the type for the sitegroup (in bytes)</description>
+        </property>
+        <property name="typesize"     type="integer"     field="type_size">
+            <description>Disk usage of the type of the sitegroup (in bytes)</description>
+        </property>
+        <property name="typerecordslimit" type="integer" field="limit_type_records">
+            <description>Limit of number of records of the type for the sitegroup</description>
+        </property>
+        <property name="typerecords" type="integer" field="type_records">
+            <description>Number of records of the type for the sitegroup</description>
+        </property>
+    </type>
+
+    <type name="midgard_activity" table="midgard_activity">
+        <property name="id" type="unsigned integer" primaryfield="id">
+            <description>Local non-replication-safe database identifier</description>
+        </property>
+        <property name="actor" link="midgard_person:id" type="unsigned integer" parentfield="actor">
+            <description>The person who performed the activity</description>
+        </property>
+        <property name="verb" type="string" index="yes">
+            <!-- Note: there must be multiple verbs when Midgard supports it -->
+            <description>The action performed, following Atom Activity Extension URL schema (for example: http://activitystrea.ms/schema/1.0/post)</description>
+        </property>
+        <property name="target" type="guid">
+            <description>The object that the action was done to</description>
+        </property>
+        <property name="summary" type="string">
+            <description>A human-readable description of the activity</description>
+        </property>
+        <property name="application" type="string" index="yes">
+            <description>Application the activity was performed with. In case of MidCOM, a component</description>
+        </property>
+    </type>
+</Schema>
--- /dev/null
+++ openpsa2-8.09.10beta1/config/midgard_auth_types.xml
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Authentication xmlns="http://www.midgard-project.org/authentication/types">
+
+  <type name="Legacy">
+    <description/>
+    <id>1</id>
+  </type>
+
+  <type name="Plaintext">
+    <description/>
+    <id>2</id>
+  </type>
+  
+  <type name="OpenID">
+    <description/>
+    <id>3</id>
+  </type>
+  
+  <type name="PAM">
+    <description/>
+    <id>4</id>
+  </type>
+  
+  <type name="Kerberos">
+    <description/>
+    <id>5</id>
+  </type>
+
+  <type name="gpg">
+    <description/>
+    <id>6</id>
+  </type>
+  
+  <type name="Shibboleth">
+    <description/>
+    <id>7</id>
+  </type>
+  
+  <type name="OAuth">
+    <description/>
+    <id>8</id>
+  </type>
+ 
+  <type name="Facebook">
+    <description/>
+    <id>9</id>
+  </type>
+
+  <type name="SHA1">
+    <description/>
+    <id>10</id>
+  </type>
+
+  <type name="SHA256">
+    <description/>
+    <id>11</id>
+  </type>
+
+  <type name="MD5">
+    <description/>
+    <id>12</id>
+  </type>
+
+  <type name="APIkey">
+    <description/>
+    <id>13</id>
+  </type>
+
+</Authentication>
--- /dev/null
+++ openpsa2-8.09.10beta1/config/midgard.conf
@@ -0,0 +1,44 @@
+
+[MidgardDatabase]
+#Database type ( by default MySQL )
+Type=SQLite
+#Name of the database
+Name=openpsa2
+#Database port ( 0 by default )
+Port=0
+#Username for user who is able to connect to database
+Username=midgard
+#Password used by user who is able to connect to database
+Password=midgard
+#Directory for SQLite database file ('~/.midgard2/data' by default)
+DatabaseDir=/var/lib/openpsa2
+#Database host ( 'localhost' by default )
+Host=localhost
+#Location of the log file
+Logfile=/var/log/openpsa2/midgard.log
+#Log level
+Loglevel=warn
+#Database creation switch
+TableCreate=true
+#Database update switch
+TableUpdate=true
+#Database and objects testing switch
+TestUnit=false
+#Midgard user's login
+MidgardUsername=admin
+#Midgard user's password
+MidgardPassword=password
+#Authentication type used with connection.
+AuthType=
+#Name of the file used with PAM authentication type
+PamFile=
+
+[MidgardDir]
+#Location of the blobs directory.
+BlobDir=/var/lib/openpsa2/blobs
+#Directory for shared, architecture independent files.
+ShareDir=/usr/share/openpsa2
+#Application specific directories.
+VarDir=/var/lib/openpsa2
+#Cached files
+CacheDir=/var/cache/openpsa2
